core/src/main/java/com/yahoo/oak/NativeMemoryAllocator.java

```diff
@@ -162,6 +162,7 @@ public void close() {
         // Release the hold of the block array and return it the provider.
         Block[] b = blocksArray;
         blocksArray = null;
+        freeList.clear();
```

Comment:
What is the purpose of the change? To release the on-heap structure? Should be cleared when entire allocator is disconnected with the OakMap/Hash

Reply:
I found this issue when I was trying the source of the memory leak in the benchmarks.
Technically, when the allocator will be collected by the GC, the `freeList` will also be collected and all its entries as well.
But it seems "more correct" to clear the list to "release" all the slices.
I'm not sure if that list is clear before. Where do we call "clear" before we call "close"? These methods are very similar.

Reply:
So this is in the hope that the garbage objects will be collected faster?

At least the comment that is couple of lines below, needs to be updated.

Reply:
:+1: